### DISCUSSION TITLE
Simplifying the many-to-many select query.

### DISCUSSION THREAD

**Discussion by DAHNEEV** [prisma/prisma#19274] [Q&A] [answered]
So I have a many-to-many relationship, Posts table, Tags and TagsOnPosts. I would like to search for posts to which the tags have specific content.
_prisma version: 4.13.0_
```
const getPosts = await prisma.posts.findMany({
        where: {
            AND: [
                {
                    tags: {
                        some: {
                            tag: {
                                content: "A"
                            }
                        }
                    }
                },
                {
                    tags: {
                        some: {
                            tag: {
                                content: "B"
                            }
                        }
                    }
                },
                            ...and so on
            ]
        }
    });
```
The input here is, for example, ["A", "B", ...].

The logical conjunction is `AND`.

The result should be posts that have tags with content "A" and "B".

**Comment by jharrell** [comment]
Hi @DAHNEEV! Wanted to make sure I understand your question:

You have `Posts` and `Tags` related via an explicit many-to-many field. You'd like to return all posts that have tags with content `"A"` and `"B"`. Is that right?

**Reply by DAHNEEV** [reply]
Yes, that's what I mean. This code what I gave above works, but if I wanted to add, for example, another argument named "C" it will be terribly long and impractical.

**Reply by ludralph** [reply, marked as answer]
Hi @DAHNEEV 

I understand your concern about the query becoming long and impractical with the addition of more arguments. 

However, a common approach to handle such situations is to dynamically construct the where clause based on the input array of tags. You would loop through each tag and add a corresponding condition to the AND array. This way, you can avoid manually writing out each condition and instead programmatically generate the query.

```js
// Assuming tagsToSearch is an array of tag contents, e.g., ["A", "B", "C"]
const tagsToSearch = ["A", "B", "C"];

const whereClause = {
  AND: tagsToSearch.map(tagContent => ({
    tags: {
      some: {
        content: tagContent,
      },
    },
  })),
};

const getPosts = await prisma.post.findMany({
  where: whereClause,
});
```
If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏
